sys/kern/imgact_elf.c

```diff
@@ -1210,9 +1210,18 @@ __CONCAT(exec_, __elfN(imgact))(struct image_params *imgp)
 				goto ret;
 			break;
 		case PT_GNU_STACK:
-			if (__elfN(nxstack))
+			if (__elfN(nxstack)) {
 				imgp->stack_prot =
 				    __elfN(trans_prot)(phdr[i].p_flags);
+#ifdef __ELF_CHERI
+				if (imgp->stack_prot & VM_PROT_EXECUTE) {
+					uprintf("CheriABI forbids executable "
+					    "stacks\n");
+					error = ENOEXEC;
+					goto ret;
+				}
+#endif
+			}
```

Comment:
Do we want to panic if `nxstack` is ever true for CheriABI? We probably want to stop you setting the tunable too.

Reply:
We could alternatively just remove the branch and supporting bits for CheriABI.  Note that it's `1` by default on all architectures support the page permission:
```C
int __elfN(nxstack) =
#if defined(__amd64__) || defined(__powerpc64__) /* both 64 and 32 bit */ || \
    (defined(__arm__) && __ARM_ARCH >= 7) || defined(__aarch64__) || \
    defined(__riscv)
        1;
#else
        0;
#endif
```

Reply:
Yeah let's just `#ifdef` the variable out for `__ELF_CHERI`, though the PT_GNU_STACK message should stay rather than silently running the binary with permissions less than requested.